src/Core/Services/TokenService.cs

```diff
@@ -209,6 +209,10 @@ public async Task<bool> GetIsExternal()
             {
                 await GetTokenAsync();
             }
+            if (_accessTokenForDecoding == null)
+            {
+                return false;
+            }
```

Comment:
IMO, even though it's something really small this should be done after the await inside the previous if, otherwise if `_accessTokenForDecoding` is not `null` at beginning it's being compared again where the result will be again not null.

```
if (_accessTokenForDecoding == null)
{
     await GetTokenAsync();
     if (_accessTokenForDecoding == null)
     {
            return false;
     }
}
```